openlibrary/accounts/model.py

```diff
@@ -658,17 +713,17 @@ def audit_accounts(email, password, test=False):
             audit['has_ia'] = ol_account.itemname
             audit['link'] = ol_account.itemname
 
-            # When enabled, this will prevent users from logging in
-            # with their Open Library credentials and instead require
-            # Internet Archive (Archive.org) credentials. 
-            # return {'error': 'ia_login_only'}
-            return audit
+            if lending.config_ia_auth_only:
+                # When enabled, this will prevent users from logging in
+                # with their Open Library credentials and instead require
+                # Internet Archive (Archive.org) credentials.
+                return {'error': 'ia_login_only'}
```

Comment:
Why wait this long to shut out OL logins when in production? In this current code, if the IA login fails, then the system still makes a db call to OL and an HTTP call to IA and *then* realizes "Oh, hey, I'm in production. I should return an error." Seems these should be moved up.